packages/fast-usdc-contract/src/exos/status-manager.ts

```diff
@@ -82,16 +59,38 @@ export const prepareStatusManager = (
   { marshaller, routeHealth }: StatusManagerPowers,
 ) => {
   /**
-   * Keyed by a tuple of the Noble Forwarding Account and amount.
+   * Keyed by Noble Forwarding Account
+   *
+   * Transactions that are `Advanced`, `AdvanceFailed`, or `AdvanceSkipped`
+   * and are awaiting a mint.
    */
-  const pendingSettleTxs: MapStore<PendingTxKey, PendingTx[]> = zone.mapStore(
+  const pendingSettleTxs: MapStore<NobleAddress, PendingTx[]> = zone.mapStore(
     'PendingSettleTxs',
     {
       keyShape: M.string(),
       valueShape: M.arrayOf(PendingTxShape),
     },
   );
 
+  const { addPendingSettleTx, matchAndDequeueSettlement } =
+    makeSettlementMatcher(pendingSettleTxs);
+
+  /**
+   * Migrate `pendingSettleTxs` to be keyed by NFA instead of the tuple of NFA
+   * and amount.
+   *
+   * Uses `zone.makeOnce` as a cheap abstraction for "only do this once".
+   */
+  zone.makeOnce('migratePendingSettleStore', () => {
+    const current = [...pendingSettleTxs.values()]; // XXX copy necessary?
+    pendingSettleTxs.clear();
+    for (const txs of current) {
+      for (const tx of txs) {
+        addPendingSettleTx(tx);
+      }
+    }
+  });
```

Comment:
Something like this may be needed more than once. Forward-compatibility suggestion derived from [`upgradeSwingset`](https://github.com/Agoric/agoric-sdk/blob/777e6a4cefd5c533aab3aeb224f95d5e1edbbd03/packages/SwingSet/src/controller/upgradeSwingset.js#L108-L352):
```suggestion
  const meta: MapStore<string, unknown> = zone.mapStore('Metadata', {
    keyShape: M.string(),
  });
  // Ensure that zone data conforms with the current schema. Historical
  // overview:
  // * version 0 (implicit): PendingSettleTxs was a
  //   MapStore<`pendingTx:${bigint}:${NobleAddress}`, PendingTx[]>
  // * version 1 (current): PendingSettleTxs is a
  //   MapStore<NobleAddress, PendingTx[]>
  const schemaVersion = Nat(meta.get('schemaVersion') || 0n);
  if (schemaVersion < 1n) {
    const oldValues = [...pendingSettleTxs.values()];
    pendingSettleTxs.clear();
    for (const txs of oldValues) {
      for (const tx of txs) {
        addPendingSettleTx(tx);
      }
    }
  }
  meta.set('schemaVersion', 1n);
```

Reply:
Awesome, this is much better. I made some tweaks to your suggestion since `meta` doesn't have a `schemaVersion` key when we call `.get()` and `.set()`